### DISCUSSION TITLE
how do i add arduino libraries? liquid crystal display for example

### DISCUSSION THREAD

**Discussion by BUGO07** [Rahix/avr-hal#520] [Q&A] [answered]
i am not sure how to import c++ libraries like ones rust uses

**Comment by djdisodo** [comment]
what lcd do you use?
at least for 1602 dot matrix display , you don't need such
there's one written for rust

**Comment by Rahix** [comment, marked as answer]
As @djdisodo wrote, don't attempt using C++ arduino libraries in Rust.  Getting this to work is _very_ complex and for most libraries, it will probably be faster to simply rewrite them in pure rust instead.  Or check the `embedded-hal` ecosystem if someone else already wrote a crate that does what you need :)

**Comment by BUGO07** [comment]
Thanks guys
